stacks/container-apps/play.prx.org.yml

```diff
@@ -201,6 +203,159 @@ Resources:
+        Variables:
+          MAX_HTTP_REDIRECTS: 4
+      Handler: index.handler
+      InlineCode: |
+        const http = require("http");
+        const https = require("https");
+
+        const PROXY_UA = "play.prx.org feed proxy";
+        const ACCEPTED_TYPES = [
+          "application/rss+xml",
+          "application/rdf+xml;q=0.8",
+          "application/atom+xml;q=0.6",
+          "application/xml;q=0.4",
+          "text/xml;q=0.4",
+        ];
+
+        async function request(url, redirectCount, redirectUrl) {
+          return new Promise((resolve, reject) => {
+            const q = new URL(redirectUrl || url);
+
+            // Setup request options
+            const options = {
+              host: q.host,
+              port: q.port,
+              path: `${q.pathname || ""}${q.search || ""}`,
+              method: "GET",
+              headers: {
+                Accept: ACCEPTED_TYPES.join(","),
+                "User-Agent": PROXY_UA,
+              },
+            };
+
+            const h = q.protocol === "https:" ? https : http;
+            const req = h.request(options, (res) => {
+              res.setEncoding("utf8");
+
+              let resData = "";
+              res.on("data", (chunk) => {
+                resData += chunk;
+                return resData;
+              });
+
+              res.on("end", async () => {
+                if (
+                  (res.statusCode >= 200 && res.statusCode < 300) ||
+                  res.statusCode === 404 ||
+                  res.statusCode === 410
+                ) {
+                  resolve(resData);
+                } else if (res.statusCode === 301 || res.statusCode === 302) {
+                  try {
+                    if (redirectCount > +process.env.MAX_HTTP_REDIRECTS) {
+                      reject(new Error("Too many redirects"));
+                      return;
+                    }
+
+                    const count = redirectCount ? redirectCount + 1 : 1;
+                    await httpRequest(url, count, res.headers.location);
+                    resolve();
+                  } catch (error) {
+                    reject(error);
+                  }
+                } else {
+                  const error = new Error(`Error ${res.statusCode}: ${resData}`);
+                  reject(error);
+                }
+              });
+            });
+
+            req.on("error", (error) => reject(error));
+
+            req.write("");
+            req.end();
+          });
+        }
+
+        exports.handler = async (event) => {
+          const feedUrl = event.queryStringParameters.url;
+
+          res = await request(feedUrl);
+
+          return {
+            statusCode: 200,
+            statusDescription: "200 OK",
+            isBase64Encoded: false,
+            headers: {
+              "Content-Type": "application/rss+xml",
+              "Cache-Control": "public, max-age=90",
+            },
+            body: res,
```

Comment:
Isn't there a limit on response sizes for API Gateway?

Reply:
I'm not sure, but there's not API Gateway here